### ISSUE TITLE
clarify hook_config_update()

### ISSUE THREAD

**Issue by docwilmot** [backdrop/backdrop-issues#765]
I thought it was called during updating a config file (ie during save) but seems it isnt? Its invoked by config_sync_file() which is only called during import.
Unless I'm way off here, this should be clarified because other _update hooks are called whenever a change happens.

**Comment by quicksketch** [comment]
Yeah all config_* hooks are called *only* during syncs, not during normal read/update/insert of config files. This is because config is a very low-level system, equivalent of doing a database query. There are many situations where modules aren't even loaded where config is read/written, so assuming that your module could interject on config updates would probably be inconsistent at best. If modules need to interject into config before it is saved, the providing module should provide a higher-level API hook for the insert/update of that object (e.g. `hook_taxonomy_vocabulary_update()` or `hook_node_type_update()`).

`hook_config_update()` itself is to respond to the updating of a config file directly, which for most modules would never happen outside of their own API. For example, you should never be modifying a node type or vocabulary by loading its config, you should be using node_type_save() or taxonomy_vocabulary_save(). I hope that all makes sense.

**Comment by docwilmot** [comment]
It does. By clarify i meant in the code comments (and hence API docs) because its not obvious.

**Comment by quicksketch** [comment]
Sorry I get all caught up explaining things. :wink:

Definitely could use more clarity in the code documentation.
